uk.co.compucorp.civicrm.hrcore/js/src/hrcore/helpers/help-icon-injector.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-env amd */
+
+(function (CRM) {
+  define(function () {
+    'use strict';
+
+    return injectHelpIcon;
+
+    /**
+     * Injects a help icon to a form field
+     *
+     * @param {jQuery} formClass - a class of the CiviCRM form
+     * @param {String} accessNameSelector - a selector to the field to populate to
+     * @param {String} title - the title of the help icon
+     * @param {String} helpId - the ID of the helper
+     * @param {String} helpFile - the path to the helper template
+     */
+    function injectHelpIcon (formClass, accessNameSelector, title, helpId, helpFile) {
```

Comment:
can't this whole function be more generic? At the end of the day, don't we want to simply append an helper icon to _some_ element? Wouldn't it be more reusable (and with a shorter signature) while still be working as expected:
```js
function injectHelpIcon (selector, icon) {
  var $helpIcon;
  var $element = $(selector);

  if (!$element.length || $element.find('.helpicon').length) {
    return;
  }

  $helpIcon = $('<a href class="helpicon" title="' + icon.title + '"></a>')
    .on('click', function () {
      CRM.help('', { id: icon.helpId, file: icon.helpFile });
    });

  // Populates a space before the icon if there are no other markers
  if (!$element.find(' .crm-marker').length) {
    $element.append('&#160;');
  }

  $element(labelSelector).append($helpIcon);
}
```
Usage:
```js
injectHelpIcon('label[for="xyz"]', {
  title: 'End Date Help',
  helpId: 'hrcareer-enddate',
  helpFile: 'CRM/HRCareer/Page/helptext'
});
```

Reply:
I don't think so. We have a real case at the moment when 2 extensions are injecting an icon into a form field label. The usage is very clean at the moment. However, as a compromise I would suggest this:

```
// helpers.js
return {
  appendHelpIconToCiviCRMFormLabel,
  appendHelpIcon
}

function appendHelpIcon (...) {
  // injects as you suggested
}

function appendHelpIconToCiviCRMFormLabel (...) {
  ... 
  appendHelpIcon(...);
}
```